google/cloud/firestore_v1/async_client.py

```diff
@@ -99,11 +99,15 @@ def _firestore_api(self):
             :class:`~google.cloud.gapic.firestore.v1`.async_firestore_client.FirestoreAsyncClient:
             The GAPIC client with the credentials of the current client.
         """
-        return self._firestore_api_helper(
+        firestore_api = self._firestore_api_helper(
             firestore_grpc_transport.FirestoreGrpcAsyncIOTransport,
             firestore_client.FirestoreAsyncClient,
             firestore_client,
         )
+        # Add _transport to FirestoreAsyncClient interface for use in Watch
+        if hasattr(firestore_api, "_client"):
+            firestore_api._transport = firestore_api._client._transport
```

Comment:
Rather than modifying the generated object instance, could we do this same logic, but in watch? If in multiple spots, maybe make an internal lookup? Or add this as a property on the client (our manual one)?

Reply:
If we create an `AsyncWatch`, this definitely something that can be moved there. In fact, I should have most likely removed this when I removed `Watch` from the async interface. I'm not quite sure what you mean by your last point - isn't this as close as we can get to adding this as a property to the manual client?

Reply:
I don't like modifying the gen object if possible. Either by adding a property (that does this) to the manual client or moving to watch would make sense to me though.

Reply:
Sounds good, will keep in mind for the `Watch` PR. Added comment to open issue.